### DISCUSSION TITLE
new architecture and activity resume

### DISCUSSION THREAD

**Discussion by adumat** [software-mansion/react-native-screens#1916] [Q&A] [answered]
Hi all,
in the new architecture is resolved the bug https://github.com/software-mansion/react-native-screens/issues/17 without the workaround of give null while onCreate of activity?

**Comment by tboba** [comment, marked as answer]
Hi @adumat, currently in the new architecture you still need to make such workaround, as React Native itself cannot recreate the fragment from savedInstanceState, as it will have to dump the whole JS virtual machine. Also, as react-native-screens creates native components, they just save unwanted things that we don't want to recreate in `onCreate` method - that's why for now you need to pass `null` to the super.onCreate.

Cheers 🤞
